src/command/verify/verify.c

```diff
@@ -2115,10 +2121,9 @@ verifyProcess(const bool verboseText)
                                                   archiveIdResult->pgWalInfo.version);
                     }
 
-                    // If we had at least one WAL range, mark all files after the last range as missing
-                    if (gapStart != NULL)
-                        verifyUpdateWalFilesMissing(jobData.backupResultList, archiveIdResult,
-                                                    gapStart, NULL, &jobData.jobErrorTotal);
+                    // Mark all files after the last range as missing
+                    verifyUpdateWalFilesMissing(jobData.backupResultList, archiveIdResult,
```

Comment:
Originally it was only called when gap was detected, not it's called always. May be make a stricter condition, with less invasive logic like:
```suggestion
          if (gapStart != NULL || rangeIdx == 0)
                    verifyUpdateWalFilesMissing(jobData.backupResultList, archiveIdResult,
```

Reply:
No, these are equivalent. `gapStart` is the first file after the last WAL range (so technically a gap, but not necessarily needed for the backup). So `gapStart` will never be NULL if we have any WAL ranges in the list.